x-pack/plugins/apm/common/projections/services.ts

```diff
@@ -16,25 +16,37 @@ import { rangeFilter } from '../utils/range_filter';
 
 export function getServicesProjection({
   setup,
+  noEvents,
 }: {
   setup: Setup & SetupTimeRange & SetupUIFilters;
+  noEvents?: boolean;
 }) {
   const { start, end, uiFiltersES, indices } = setup;
 
   return {
-    index: [
-      indices['apm_oss.metricsIndices'],
-      indices['apm_oss.errorIndices'],
-      indices['apm_oss.transactionIndices'],
-    ],
+    ...(noEvents
+      ? {}
+      : {
+          index: [
+            indices['apm_oss.metricsIndices'],
+            indices['apm_oss.errorIndices'],
+            indices['apm_oss.transactionIndices'],
+          ],
+        }),
```

Comment:
will this be changed by your PR for new client?

Reply:
yep! there will no longer be a need for `noEvents`.